File: scala/Flags.scala
import scala.collection.breakOut

case class Flags(exename: String, compiler: String, fcompile: String, flink: String)
object Flags
{
    def apply(zipname: String, args: Array[String]): Flags =
    {
        val map: Map[String, String] = (for (arg <- args; if arg.contains("="); (key, value) = arg.span(_!='=')) yield key -> value.tail)(breakOut)
        val exename = map.getOrElse("exename", zipname.split('.').head)
        // If the compiler isn't specified we should default to g++ to be on the safe side
        val compiler = map.getOrElse("compiler", "g++")
        val fcompile = map.getOrElse("compiler-flags", "")
        val flink = map.getOrElse("linker-flags", "")
        new Flags(exename, compiler, fcompile, flink)
    }
}

File: python/flags.py
class Flags:
    def __init__(self, zipname, *args):
        m = dict()
        for arg in args:
            if "=" not in arg: continue
            key, value = arg.split("=", 1)
            m[key] = value
        self.exename = m["exename"] if "exename" in m else zipname.split(".")[0]
        self.compiler = m["compiler"] if "compiler" in m else "g++"
        self.fcompile = m["compiler-flags"] if "compiler-flags" in m else ""
        self.flink = m["link-flags"] if "link-flags" in m else ""
